modulemd/v2/include/modulemd-2.0/private/modulemd-yaml.h

```diff
@@ -421,4 +421,129 @@ modulemd_yaml_emit_document_headers (yaml_emitter_t *emitter,
+    }                                                                         \
+  while (0)
+
+#define EMIT_HASHTABLE_VALUES(emitter, error, key, table, emitfn)             \
+  do                                                                          \
+    {                                                                         \
+      if (NON_EMPTY_TABLE (table))                                            \
+        {                                                                     \
+          EMIT_KEY (emitter, error, key);                                     \
+          EMIT_MAPPING_START (emitter, error);                                \
+          gsize i;                                                            \
+          g_autoptr (GPtrArray) keys =                                        \
+            modulemd_ordered_str_keys (table, modulemd_strcmp_sort);          \
+          for (i = 0; i < keys->len; i++)                                     \
+            {                                                                 \
+              if (!emitfn (                                                   \
+                    g_hash_table_lookup (table, g_ptr_array_index (keys, i)), \
+                    emitter,                                                  \
+                    error))                                                   \
+                return FALSE;                                                 \
+            }                                                                 \
+          EMIT_MAPPING_END (emitter, error);                                  \
+        }                                                                     \
+    }                                                                         \
+  while (0)
+
+#define EMIT_HASHTABLE_KEY_VALUES(emitter, error, key, table)                 \
+  do                                                                          \
+    {                                                                         \
+      if (NON_EMPTY_TABLE (table))                                            \
+        {                                                                     \
+          EMIT_KEY (emitter, error, key);                                     \
+          EMIT_MAPPING_START (emitter, error);                                \
+          gsize i;                                                            \
+          g_autoptr (GPtrArray) keys =                                        \
+            modulemd_ordered_str_keys (table, modulemd_strcmp_sort);          \
+          for (i = 0; i < keys->len; i++)                                     \
+            {                                                                 \
+              EMIT_KEY (emitter, error, g_ptr_array_index (keys, i));         \
+              EMIT_KEY (                                                      \
```

Comment:
Yeah, this is another place where calling it `EMIT_SCALAR()` makes more sense. It reads confusingly to `EMIT_KEY(key)` and `EMIT_KEY(value)` together.